zkvm/spec/ZkVM.md

```diff
@@ -1248,14 +1252,23 @@ Pushes an [expression](#expression-type) `expr` corresponding to the [maximum ti
 
 The one-term expression represents time bound as a weight on the R1CS constant `1` (see [`const`](#const)).
 
+#### expr
+
+_var_ **expr** → _ex_
+
+1. Pops a [variable](#variable-type) `var`.
+2. If it is [detached](#attached-and-detached-variables), attaches it to the constraint system.
```

Comment:
If the variable is already attached, does `expr` do nothing?

Reply:
`expr` creates Expression object, consuming Variable object regardless. But it's not really much of a side-effect - one can drop and dup Variables at will.